includes/Plugin.php

```diff
@@ -140,6 +142,20 @@ class Plugin {
 	 */
 	public $experiments;
 
+	/**
+	 * AMP plugin integration.
+	 *
+	 * @var AMP
+	 */
+	public $amp;
```

Comment:
I was thinking about this and having to many fields on a class. I how about idea. 

```
public $integrations = [];
```
then 
```
$this->integrations['amp'] = new AMP();
```

We would then break up this class into array of instances based on Namespace. integrations property for classes in the integrations namaspace etc. Then use a magic getting for backwards compatibility?

Reply:
Yeah was basically thinking of the same; either 

```php
public $integrations = [];

$this->integrations['amp'] = new AMP();
```

or something like this

```php
public $integrations = new Integrations(); // or new stdObject

$this->integrations->amp = new AMP();
```

Everything beyond that (splitting up this class, magic getter, backwards compatibility, etc.) can be looked at in #5062